### ISSUE TITLE
rp-hal

### ISSUE THREAD

**Issue by charles-d-burton** [henrikssn/rust-radio-sx1231#6]
I'm pretty new to interfacing hardware with rust and I was wondering if this would work with the rp-hal project?  It has a generic hal interface for the rp2040 board and I would very much like to interface one of these module with the board.  Ideally I want to set one as a base station on a raspberry pi and then have multiple rp2040 based sensors sending data to the base station.  It's mostly an experiment in learning how this is all supposed to work and learn rust along the way.  It seems like it should be possible considering that I can get the SPI interface from the rp-hal and then ideally send that interface to your library to interface the two but I'm not really sure where to begin.

https://github.com/rp-rs/rp-hal

**Comment by henrikssn** [comment]
Yes, essentially passing an Spi object and a CS pin to `Sx1231::new` is the way to go. Give it a try and post some code with error messages if you get stuck.
